### ISSUE TITLE
Tracery1 delayed expansion syntax resulting in SVG render issue (and also, ALT text truncation)

### ISSUE THREAD

**Issue by alien-sunset** [BooDoo/CheapBotsTootSweet#27]
i don't know where the issue for this lies... perhaps I'm using a newer version of tracery (I've got "tracery-grammar": "^2.7.4" on my node) (which is impressive considering how old it is)
or a disconnect or some odd incompatibility re: browsers.

but i have some code (up on [AlienSunsets_TestBot](https://botsin.space/@AlienSunsets_TestBot)) and the SVG isn't working.
but on my home machine with no changes it works fine

I've spent hours trying to figure this out, and I'm just at a loss.

![image](https://github.com/user-attachments/assets/fe967d09-62d8-440d-ba5f-cc4d3e626632)
![image](https://github.com/user-attachments/assets/e02a69a7-075e-4898-b36e-dec73db842bf)
![image](https://github.com/user-attachments/assets/3d1ea7f5-1aca-4d86-9542-4a8a6c71314b)

**Comment by alien-sunset** [comment]
made a glitch project to show the code:
I've tried with both versions of the tracery and neither works on CBTS

peeking at the page source on cbts, it looks like for some reason the "#" symbols necessary for the color hex codes and pattern id urls is being completely stripped away 

https://glitch.com/edit/#!/aliensaliens

**Comment by alien-sunset** [comment]
i was semi able to fix the issue above by fiddling with how things were nested, but I'm coming up against a similar issue with another bot, where it's not able to read items within nested rules, and i'm not entirely sure i'm going to be able to un nest them enough to get it to work properly.

i really do think that you need to fully update to teh most recent tracery version.
(which really, isn't all that recent tbh and probably should have been done a long time ago....)

**Comment by alien-sunset** [comment]
i've updated the glitch linked above to have the Socks too

this is what that code looks like on CBTS for me:
![image](https://github.com/user-attachments/assets/c7a0051e-52da-4330-bd1f-1fe8d6f21cdf)

and when posted:
![image](https://github.com/user-attachments/assets/57c3b267-c665-43f6-97b1-77d2edc1312c)

when you can see on the glitch, the alt text all parses fine.

i said this on Mastodon, but i'm going to copy paste it here for record keeping/continuity:

It seems to be a similar issue to the previous one, where nested rules are being read wrongly 

Fir the aliens it Was prematurely un escaping the hashes in the color hexcodes and then trying to read them as tracery code, and for the socks it’s flattening a string in a ## early and trying to parse it like tracery code,  which is resulting in half the sentence being lost.

"singleorigin" works perfectly for me too, but it's calling "singlesocks" which is using "centralbox" and "oneSoxAlt" directly

for "origin" i wanted it to pick from three possibilities which each needed their own separate alts so it's pulling from  "picksocks" which is choosing between  "twoSox" "twoDiffSox" and  "oneSox"

which have a nested rule:
        "twoSox":["[soxalt:#twoSoxAlt#][sox:[#sock1Box]##otherSock1Box#]"],
        "twoDiffSox":["[soxalt:#twoDiffSoxAlt#][sox:#sock1Box##sock2Box#]"],
        "oneSox":["[soxalt:#oneSoxAlt#][sox:#centralBox#]"],

and i think it's trying to flatten(expand?) the alts BEFORE calling "soxalt" and then parsing it lake tracery which is splitting up the sentences at the commas

so  instead of going #picksocks# = "twoSox" so #soxalt# = #twoSoxAlt# = "#base2Color.a# sock with #heel2Color.a# heel, #toe2Color.a# toe, and #cuff2Color.a# cuff, with #decoName2#"

it's going [soxalt:#base2Color.a# sock with #heel2Color.a# heel, #toe2Color.a# toe, and #cuff2Color.a# cuff, with #decoName2#]

and with nested rules you don't need quotes, it just parses commas as separating the choices: so it thinks it's looking at ...

#soxalt# = "#base2Color.a# sock with #heel2Color.a# heel",
" #toe2Color.a# toe",
"and #cuff2Color.a# cuff",
"with #decoName2#"

and then choosing randomly between those

at least, that's the best I can figure from what I see happening.

and like i said, this exact code all works fine in VSCode and on BBDQ (and glitch)

**Comment by BooDoo** [comment]
Based on testing, my understanding of the more recent case (ALT text generation for socks returning randomized/truncated value) is this resulted from the frontend using legacy Tracery module. The behavior you intended is what I referred to as "delayed expansion" in Issue #16. I suspect your original alien code had a similar issue since you say it was resolved by fiddling with the nesting.

Your grammar as-written should have posted the expected output when generated on the backend/schedule process, but both the in-browser preview **as well as the `send_status` button in the web client** could run into the problem you were seeing.

BooDoo/traceryhosting-frontend@acb4cc7 should align frontend and `send_status` behavior with the scheduled `run_bots.js` parsing. Preliminary testing against your specific socks issue and against the documented Tracery2-friendly format for embedding braces within meta objects (one of the blockers for the original front-end migration) look like we're in a good place now.
